src/backend/distributed/commands/role.c

```diff
@@ -124,6 +124,13 @@ RoleSpecToObjectAddress(RoleSpec *role, bool missing_ok)
 		Oid roleOid = get_rolespec_oid(role, missing_ok);
 		ObjectAddressSet(*address, AuthIdRelationId, roleOid);
 	}
+	else
+	{
+		/*
+		 * If rolespec is null, role can be 'ALL'. We should be returning a pseudo-valid oid.
```

Comment:
The problem is tricky. What if we return an empty list? Even if we can assume `OID_MAX` is can be used by us, we should adjust all the callers as well. Not sure if we can do that properly.

I think the best way would be leave this function as-is, and consider adding an exception in `DistOpsHasInvalidObject` for `ALTER ROLE ALL`. Can we do that?

Reply:
moved this discussion to the original PR #6114

Reply:
I closed it and resolved it here.